### DISCUSSION TITLE
How can I import a package inside another one

### DISCUSSION THREAD

**Discussion by ValentinH** [belgattitude/nextjs-monorepo-example#1333] [Q&A] [answered]
I'd like to know how would you do to use a package inside another one. For example, using a function from `core-utils` inside `ui-utils`?

**Comment by belgattitude** [comment, marked as answer]
Here's a reply with an example here https://github.com/belgattitude/nextjs-monorepo-example/pull/1347. It basically adds the @your-org/core-lib dependency to the @your-org/ui-lib


<img src="https://user-images.githubusercontent.com/259798/154331551-7a6afaa1-7a3f-4391-ba6a-0c7b8aa8bfb5.png" width="350" />

Interesting parts are:

1. state your new dep, in dependencies (or peerDependencies) 

```json5
  "dependencies": {
    "@your-org/core-lib": "workspace:^"
  },
```

2. inform typescript how to resolve it in tsconfig.json

```json5

{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "baseUrl": "./src",
    "paths": {
      // First line allows you to resolve `import {}  from "@your-org/core-lib/folder1/subfolder/file"`
      "@your-org/core-lib/*": ["../../../packages/core-lib/src/*"],
      // Second line resolves the barrel. `import {}  from "@your-org/core-lib"` 
      "@your-org/core-lib": ["../../../packages/core-lib/src/index"]
    }
  }
}
```

PS : Exposing the barrel (index.ts) or the modules exports is preferred if you intend to publish the package... 


Let me know if you want more in-depth and hope it will help you

**Reply by ValentinH** [reply]
Thanks a lot! I actually did this but had an issue with Storybook. However, you probably had it as well and solved it in #1347. This fix was also what I need.

All good now.

Thanks again for maintaining this very cool example!

**Reply by belgattitude** [reply]
Yep storybook lags a bit. In the the near future I think those tricks won't be needed. I'm really looking forward to v7 and full webpack 5 support.
